src/main/java/com/google/devtools/build/lib/sandbox/SandboxStash.java

```diff
@@ -61,30 +83,57 @@ public class SandboxStash {
 
   private final Map<Path, String> stashPathToRunfilesDir = new ConcurrentHashMap<>();
 
-  public SandboxStash(String workspaceName, Path sandboxBase) {
+  private static final int POOL_SIZE = Runtime.getRuntime().availableProcessors();
+  private final ExecutorService stashFileListingPool =
+      Executors.newFixedThreadPool(
```

Comment:
Have you tried/considered using virtual threads here? This looks mostly IO-bound.

Reply:
Makes sense to do this in a follow-up change (although I think that the compensation mechanism has been implemented for vfs).